### PR TITLE
Functions now accept Body arguments

### LINKED ISSUE #21964: Make `Body` and `ReferenceFrame` analogous
`Body` and `ReferenceFrame` are analogous in dynamics, and all the joint classes , jointsmethod and class body follow it, but rest of physics module doesn't.

Example - 

```
>>> A = Body('A')
>>> N = ReferenceFrame('N')
>>> A.dcm(N) # Works
>>> A.dcm(A) # Works

>>> N.dcm('A') # Error but should work

```

So where ever ReferenceFrame works on being passed as arg, passing body to same arg should also work.

It can be done simply by - 
```
if isinstance(frame, Body):
    frame = Body.frame
```

cc - @moorepants

### PR DESCRIPTION
**Pull request by mohajain**
#### References to other Issues or PRs

Fixes #21964
#### Brief description of what is fixed or changed

functions now accept body arguments along with reference frame
#### Release Notes

* physics.mechanics
  * Added the option to pass Body as an argument where RefrenceFrame is used

### CODE REVIEW

**Comment by mostlyaman** [comment]
Please refrain from deleting PRs. Most issues can be fixed without the need to close the PR. I am guiding you to solve the problem in a bit.

**Comment by mohajain** [comment]
I am sorry for the inconvenience caused. Please guide me through this.

**Comment by mostlyaman** [comment]
There is no need to apologize.😅 I am also a fairly new recent contributor
so I know sometimes it is terrifying when something doesnt work. I am
framing a solution for you. please give me a min.
Best Regards,
Aman Sharma

File: .mailmap [hunk 1]
```diff
@@ -583,3 +583,5 @@ Rahil Parikh <r.parikh@somaiya.edu> rprkh <r.parikh@somaiya.edu>
 Yaser AlOsh <yaseralosh@outlook.com> Yaser <yaseralosh@outlook.com>
 Joannah Nanjekye <joannah.nanjekye@ibm.com> nanjekyejoannah <joannah.nanjekye@ibm.com>
 Kshitij <kshitijparwani.mat18@itbhu.ac.in> Kshitij Parwani <44468674+P-Kshitij@users.noreply.github.com>
+Moha Jain <mohajain99@gmail.com> mohajain <mohajain99@gmail.com>
+Moha Jain <mohajain99@gmail.com> mohajain <45903778+mohajain@users.noreply.github.com>
```
Comment (sidhu1012) [inline .mailmap:587]: This doesn't seem right.

File: sympy/physics/mechanics/functions.py [hunk 2]
```diff
@@ -81,7 +81,9 @@ def inertia(frame, ixx, iyy, izz, ixy=0, iyz=0, izx=0):
     (N.x|N.x) + 2*(N.y|N.y) + 3*(N.z|N.z)
 
     """
-
```
Comment (sidhu1012) [inline sympy/physics/mechanics/functions.py:84]: Dont remove this blank line. Fic it everywhere.

File: .mailmap (hunk 1 above)
Comment (mohajain) [reply .mailmap:587]: I simply followed the steps mentioned above. What's wrong with this?

**Comment by mohajain** [comment]
Thank you so much @mostlyaman for helping me through this!

**Comment by sidhu1012** [comment]
This PR still needs tests and updated docstrings.

File: .mailmap (hunk 1 above)
Comment (mostlyaman) [reply .mailmap:587]: @sidhu1012 Please tell me if there is some problem with this. I was added in AUTHORS when oscarbenjamin was updating the authors and mailmap and he added my entries in the same way. I have also read through the instructions in mailmap and authors file and I believe this is correct although I may be wrong. Maybe oscarbenjamin can confirm if I am wrong.

**Comment by mohajain** [comment]
> This PR still needs tests and updated docstrings.

What should be added in documentation and tests?

**Comment by sidhu1012** [comment]
> What should be added in documentation and tests? 
> 

Tests to check the changes are working correctly and don't break any code.

Update docstrings to make user know that they can pass body as an arg too.

**Comment by mohajain** [comment]
> Update docstrings to make user know that they can pass body as an arg too.

Do I have to add the documentation in the parameters section of each function i changed?

**Comment by sidhu1012** [comment]
> Do I have to add the documentation in the parameters section of each function i changed?

Update the documentation in parameters section and add tests is corresponding test file.

**Comment by mohajain** [comment]
While making the tests I found out that the `sympy.physics.vector.point` file also has these functions which take reference frames as arguments. Please have a look and tell me if I should change them as well.

**Comment by sidhu1012** [comment]
Yes they should.

**Comment by mohajain** [comment]
I am getting this error while adding tests for functions.py
```
File "/home/mj/Documents/sym/sympy/sympy/physics/mechanics/tests/test_functions.py", line 98, in test_linear_momentum
    raises(TypeError, lambda: linear_momentum(N, N, Pa))
  File "/home/mj/Documents/sym/sympy/sympy/testing/pytest.py", line 104, in raises
    raise Failed("DID NOT RAISE")
sympy.testing.pytest.Failed: DID NOT RAISE
```

**Comment by sidhu1012** [comment]
The error you are trying to catch isn't being raised.

**Comment by mohajain** [comment]
How can I fix this? Should I just remove this raise line from the test?

**Comment by mostlyaman** [comment]
You could replace that line with something like
```python
assert linear_momentum(N, N, Pa) == 10 * N.x
```

**Comment by sidhu1012** [comment]
Please resolve conflicts first so that tests could run.

**Comment by mohajain** [comment]
```
File "/home/mj/Documents/sym/sympy/sympy/physics/vector/tests/test_frame.py", line 196, in test_dcm
    A = N.orientnew('A', 'Axis', [q1, N.z])
AttributeError: 'Body' object has no attribute 'orientnew'
```
I am getting this error while making tests for frame.py as the Body class doesn't have orientnew. What should be done here?

**Comment by mohajain** [comment]
@sidhu1012

**Comment by sidhu1012** [comment]
As I said before, please resolve conflicts so that I can see the test failures and tests could run.

File: .mailmap (hunk 1 above)
Comment (oscarbenjamin) [reply .mailmap:588]: This looks fine to me

**Comment by sidhu1012** [comment]
> ```
> File "/home/mj/Documents/sym/sympy/sympy/physics/vector/tests/test_frame.py", line 196, in test_dcm
>     A = N.orientnew('A', 'Axis', [q1, N.z])
> AttributeError: 'Body' object has no attribute 'orientnew'
> ```
> I am getting this error while making tests for frame.py as the Body class doesn't have orientnew. What should be done here?

Body isn't supposed to support `orient` and  `orient_new()`.

**Comment by mohajain** [comment]
> Body isn't supposed to support `orient` and `orient_new()`.

How should I make tests for dcm?
I was using the testcase already given and changing the reference frame argument with a body argument
```
    N = Body('N')
    A = N.orientnew('A', 'Axis', [q1, N.z])
    B = A.orientnew('B', 'Axis', [q2, A.x])
    C = B.orientnew('C', 'Axis', [q3, B.y])
    D = N.orientnew('D', 'Axis', [q4, N.y])
    E = N.orientnew('E', 'Space', [q1, q2, q3], '123')
    assert N.dcm(C) == Matrix([
        [- sin(q1) * sin(q2) * sin(q3) + cos(q1) * cos(q3), - sin(q1) *
        cos(q2), sin(q1) * sin(q2) * cos(q3) + sin(q3) * cos(q1)], [sin(q1) *
        cos(q3) + sin(q2) * sin(q3) * cos(q1), cos(q1) * cos(q2), sin(q1) *
            sin(q3) - sin(q2) * cos(q1) * cos(q3)], [- sin(q3) * cos(q2), sin(q2),
        cos(q2) * cos(q3)]])
```

**Comment by sidhu1012** [comment]
This requires release notes.

**Comment by mohajain** [comment]
Is it okay now?

**Comment by sidhu1012** [comment]
I'll look at it tomorrow.

**Comment by sidhu1012** [comment]
It also needs tests for changes in `vector.py`, else lgtm.

**Comment by mohajain** [comment]
I haven't made any changes to `vector.py`. What tests should I add? @sidhu1012

**Comment by sidhu1012** [comment]
I do see changes in physics/vector/frame.pyvand point.py but you haven't added tests for these changes.

**Comment by mohajain** [comment]
The tests for frame.py and point.py arent working as the tests already present use functions like `orient_new()` and `set_ang_vel()` which are not available for Body.

File: sympy/physics/vector/frame.py [hunk 3]
```diff
@@ -387,8 +388,8 @@ def ang_acc_in(self, otherframe):
         Parameters
         ==========
 
-        otherframe : ReferenceFrame
-            The ReferenceFrame which the angular acceleration is returned in.
+        otherframe : ReferenceFrame or Body
+            The ReferenceFrame or Body which the angular acceleration is returned in.
```
Comment (sidhu1012) [inline sympy/physics/vector/frame.py:393]: These changes should be done in public function where it is called, this function should receive frame only, unless there is an issue with it.

**Comment by sidhu1012** [comment]
> The tests for frame.py and point.py arent working as the tests already present use functions like `orient_new()` and `set_ang_vel()` which are not available for Body.

Tests for changes in `frame.py` and `orient.py` should work fine. You should add new tests for this new functionality.

File: sympy/physics/vector/frame.py (hunk 3 above)
Comment (mohajain) [reply sympy/physics/vector/frame.py:393]: > These changes should be done in public function where it is called, this function should receive frame only, unless there is an issue with it.

I am not able to understand this, the function `ang_acc_in()` in which this change is made is already public. In which functions do I have to make changes?

**Comment by mohajain** [comment]
> > The tests for frame.py and point.py arent working as the tests already present use functions like `orient_new()` and `set_ang_vel()` which are not available for Body.
> 
> Tests for changes in `frame.py` and `orient.py` should work fine. You should add new tests for this new functionality.

I haven't made any changes in `orient.py`. In `frame.py`, as I stated before, the tests that are already present for ReferenceFrame arguments use functions specific to ReferenceFrame like `orient_new()` so I cannot make a test case using the same test but with Body as an argument.

**Comment by mohajain** [comment]
@sidhu1012 could you please help me?

**Comment by sidhu1012** [comment]
@mohajain Can you please fix conflicts so that tests could run again.

**Pull request closed without merging**